frontend/src/pages/eventOverview.js

```diff
@@ -92,87 +93,56 @@ class EventOverview extends Component {
-                      onChange={this.handleChange}
-                      onKeyPress={this.handleKeyPress}
-                    />
-                  </FormGroup>
-                  <FormGroup>
-                    <Label>Description</Label>
-                    <Input
-                      name="description"
-                      type="textarea"
-                      placeholder="i.e. Teams present their final product."
-                      onChange={this.handleChange}
-                      onKeyPress={this.handleKeyPress}
-                    />
-                  </FormGroup>
-                  <FormGroup>
-                    <Label>Facebook Event Link</Label>
-                    <Input
-                      name="fbLink"
-                      type="url"
-                      placeholder="i.e. https://www.facebook.com/events/2405129342892922/"
-                      onChange={this.handleChange}
-                      onKeyPress={this.handleKeyPress}
-                    />
-                  </FormGroup>
-                </Form>
-              </ModalBody>
-              <ModalFooter>
-                <Button color="secondary" onClick={this.toggleModal}>
-                  Cancel
-                </Button>
-                <Button color="primary" onClick={this.addEvent}>
-                  Submit
-                </Button>
-              </ModalFooter>
-            </Modal>
+            <EventsModal
+              title="Add New Event"
+              isOpen={this.state.modal}
+              formFields={[
+                {
+                  label: 'Name',
+                  name: 'name',
+                  type: '',
+                  placeholder: 'i.e. Product Showcase'
+                },
+                {
+                  label: 'Date',
+                  name: 'date',
+                  type: 'date',
+                  placeholder: ''
+                },
+                {
+                  label: 'Start Time',
+                  name: 'startTime',
+                  type: 'time',
+                  placeholder: ''
+                },
+                {
+                  label: 'End Time',
+                  name: 'endTime',
+                  type: 'time',
+                  placeholder: ''
+                },
+                {
+                  label: 'Location',
+                  name: 'location',
+                  type: '',
+                  placeholder: 'i.e. ECEB'
+                },
+                {
+                  label: 'Description',
+                  name: 'description',
+                  type: 'textarea',
+                  placeholder: 'i.e. Teams present their final product.'
+                },
+                {
+                  label: 'Facebook Event Link',
+                  name: 'fbLink',
+                  type: 'url',
+                  placeholder: 'i.e. https://www.facebook.com/events/2405129342892922/'
+                }
+              ]}
```

Comment:
I like this, but I would move this to a different file (in utils or something), and then import it and pass the the constant variable set to this JSON array.